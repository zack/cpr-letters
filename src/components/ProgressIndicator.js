import React from 'react';
import { ProgressBar } from 'react-bootstrap';

const ProgressIndicator = ({green, yellow, red}) => {
  const total = green + red + yellow;
  return(
    <ProgressBar className="stamp">
      <ProgressBar bsStyle="success" max={total} label={green} now={green} key={1} />
      <ProgressBar bsStyle="warning" max={total} label={yellow} now={yellow} key={2} />
      <ProgressBar bsStyle="danger" max={total} label={red} now={red} key={3} />
    </ProgressBar>
  );
}

export default ProgressIndicator;
