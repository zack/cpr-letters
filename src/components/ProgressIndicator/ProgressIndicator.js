import React from 'react';
import { ProgressBar } from 'react-bootstrap';

const ProgressIndicator = ({green, yellow, red}) => {
  const total = green + red + yellow;
  const green_string = green ? `${Math.ceil(green/total*100)}%` : '';
  const yellow_string = yellow ? `${Math.ceil(yellow/total*100)}%` : '';

  return(
    <ProgressBar className="stamp">
      <ProgressBar bsStyle="success" max={total} label={green_string} now={green} key={1} />
      <ProgressBar bsStyle="warning" max={total} label={yellow_string} now={yellow} key={2} />
    </ProgressBar>
  );
}

export default ProgressIndicator;
