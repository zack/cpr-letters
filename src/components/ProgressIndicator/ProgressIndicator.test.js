import React from 'react';
import ProgressIndicator from './ProgressIndicator.js';
import renderer from 'react-test-renderer';

describe('<ProgressIndicator />', () => {
  it('Reflects the green, yellow, and red bars', () => {
    [[0,0,0], [0,0,1], [0,1,0], [0,1,1], [1,0,0], [1,0,1], [1,1,0], [1,1,1]]
    .forEach((numbers) => {
      const [green, yellow, red] = numbers;
      const component = renderer.create(
        <ProgressIndicator green={green} yellow={yellow} red={red} />
      );
      expect(component).toMatchSnapshot();
    });
  });
});
