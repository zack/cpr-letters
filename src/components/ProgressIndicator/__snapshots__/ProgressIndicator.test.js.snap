// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 1`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={0}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  >
    
  </div>
  <div
    aria-valuemax={0}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  >
    
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 2`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 3`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    100%
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 4`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 5`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    100%
  </div>
  <div
    aria-valuemax={1}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 6`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 7`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
  <div
    aria-valuemax={2}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`<ProgressIndicator /> Reflects the green, yellow, and red bars 8`] = `
<div
  className="stamp progress"
>
  <div
    aria-valuemax={3}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "33.333%",
      }
    }
  >
    34%
  </div>
  <div
    aria-valuemax={3}
    aria-valuemin={0}
    aria-valuenow={1}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "33.333%",
      }
    }
  >
    34%
  </div>
</div>
`;
